src/main/java/de/rub/nds/scanner/core/probe/ScannerProbe.java

```diff
@@ -282,155 +100,49 @@ public final void putIfEqual(
         }
     }
 
-    public final void putIfEqual(
-            AnalyzedProperty determiningProperty,
-            AnalyzedProperty propertyToSet,
-            Map<?, ?> actualResult,
-            String notApplicableReason,
-            TestResult expectedValue) {
-        if (!propertiesMap.containsKey(determiningProperty)
-                || propertiesMap.get(determiningProperty) == null
-                || !propertiesMap.get(determiningProperty).equals(expectedValue)) {
-            put(propertyToSet, new NotApplicableResult(propertyToSet, notApplicableReason));
-        } else {
-            put(propertyToSet, actualResult);
+    private TestResult convertToResult(AnalyzedProperty property, Object result) {
+        // TODO: Use pattern matching with java 17(+)
+        if (result instanceof TestResult) {
+            return (TestResult) result;
         }
-    }
-
-    public final void putIfEqual(
-            AnalyzedProperty determiningProperty,
-            AnalyzedProperty propertyToSet,
-            Set<?> actualResult,
-            String notApplicableReason,
-            TestResult expectedValue) {
-        if (!propertiesMap.containsKey(determiningProperty)
-                || propertiesMap.get(determiningProperty) == null
-                || !propertiesMap.get(determiningProperty).equals(expectedValue)) {
-            put(propertyToSet, new NotApplicableResult(propertyToSet, notApplicableReason));
-        } else {
-            put(propertyToSet, actualResult);
+        if (result instanceof String) {
+            return new StringResult(property, (String) result);
         }
```

Comment:
Why not simply maintain a static dictionary to match value and test result class? This has several advantages:

- Iteration possible (less verbose code)
- Registration of custom test result classes becomes possible by adding to the dictionary without overriding the method

Reply:
This function is just for convenience to map int to IntegerResult (...).
- The order of comparison matters (TestResult needs to be first, Obejct last)
- custom result classes (as long as they inherit TestResult, which is a quite lenient interface) are returned as such and not wrapped. Custom Result classes now don't have the benefit of mapping `Custom` to `CustomResult`, which I think is fine.
    - This way, the CustomResult constructor is also free to accept any arguments you wish for, instead of just `property, result`.